src/Engine/Rendering/Renderer.cpp

```diff
@@ -714,6 +714,11 @@ int Renderer::buildAllRenderTechniques() const {
     return 0;
 }
 
+Scalar Renderer::getDepth( int, int ) {
+    CORE_ASSERT( false, "Renderer::getDepth() not implemented, must be overriden" );
```

Comment:
as the (abstract) class `Renderer` own a z-buffer (the member `m_depthTexture`) This method could be implemented using this depth buffer. But you should add a note in the documentation saying that, if a concrete renderer do not use the base depth-buffer, this method should be overridden. it is the case, e.g. for Forward Renderer that use its own depth-buffer for scene rendering.

Another solution, instead of an error message, is tu make this method pure virtual so that any renderer must implement it.

Reply:
Done, but reading a texture (without fbo) needs to read back the whole z buffer.
Moreover ForwardRenderer do not use m_depthTexture.
I'm not sure what is the better.